### PR TITLE
fix(gomod): Now ignoring multi-line replace directive inside go.mod files.

### LINKED ISSUE #11428: A `replace` directive in a `go.mod` won't be ignored if it's a multi-line block (while single-liners are ignored as expected)
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

Bitbucket Cloud (bitbucket.org)

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v26.6.0

### Describe the bug

(Probably related to https://github.com/renovatebot/renovate/issues/6647)

A `replace` directive in a `go.mod` file can either be a single-liner or span multiple lines. Both of the following are valid `replace` directives. (See https://golang.org/ref/mod#go-mod-file-replace for further information.)
```
replace example.com/module_a => ../module_a

replace (
    example.com/module_a => ../module_a
    example.com/module_b => ../module_b
)
```

However, it seems that Renovate is only capable of handling the single-line case, as is suggested by the following code:
https://github.com/renovatebot/renovate/blob/d260dbf2d970b2d097b45b59cb3067ac532916a7/lib/manager/gomod/artifacts.ts#L109-L112

In the multi-line case Renovate will fail to run `go get -d ./...` because the `replace`-ments are not being ignored.

I've created the following repositories (on GitHub though) to demonstrate the issue.
- https://github.com/renovate-gomod-issue/foo, a module having tags v1.0.0 and v1.0.1.
- https://github.com/renovate-gomod-issue/bar, another module having tags v1.0.0 and v1.0.1.
- https://github.com/renovate-gomod-issue/partially-replaced, a module depending on `foo` v1.0.0 and `bar` v1.0.0, where `foo` is replaced.
- https://github.com/renovate-gomod-issue/fully-replaced, another module depending on `foo` v1.0.0 and `bar` v1.0.0, where both `foo` and `bar` are replaced.

I've [forked and edited Renovate](https://github.com/renovate-gomod-issue/renovate/pull/1/files#diff-b53c48291df312e71cae3acd27c0f571c2d6c0234a4f2444e0293196a4d76f05) so that it prints out the content of any `go.mod` before and after the "massaging".

When I ran `RENOVATE_TOKEN=xxx npm start` with [the config here](https://github.com/renovate-gomod-issue/renovate/pull/1/files#diff-e8e9eea3276b2de9a74fd257902dc74c846f6fc8653bbddf45d738b71201f93d), Renovate succeeded with the `partially-replaced` repository and failed with the `fully-replaced` repository.

The complete logs are included in the branch as well:
- [stdout](https://github.com/renovate-gomod-issue/renovate/pull/1/files#diff-275f8eb0b3670719242ead5792bf63d698bda7b3a0506a3a7d2880a8b44fecb9)
- [renovate.log](https://github.com/renovate-gomod-issue/renovate/pull/1/files#diff-17ca9073197e56ebb6828c2911e7aeb983bc6c17ec780ea4cf6305df3ec1545c)

### Relevant debug logs

<details><summary>Logs</summary>

```
 INFO: Lock file error - aborting (repository=renovate-gomod-issue/fully-replaced)

{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","msg":"gomod.updateArtifacts(go.mod)","time":"2021-08-25T14:22:11.336Z","v":0}
{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","msg":"Compare go.mod before and after massaging","time":"2021-08-25T14:22:11.337Z","v":0}
{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","msg":"module github.com/renovate-gomod-issue/fully-replaced\n\ngo 1.16\n\nrequire (\n\tgithub.com/renovate-gomod-issue/bar v1.0.1\n\tgithub.com/renovate-gomod-issue/foo v1.0.0\n)\n\nreplace (\n\tgithub.com/renovate-gomod-issue/bar => ../bar\n\tgithub.com/renovate-gomod-issue/foo => ../foo\n)\n","time":"2021-08-25T14:22:11.337Z","v":0}
{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","msg":"module github.com/renovate-gomod-issue/fully-replaced\n\ngo 1.16\n\nrequire (\n\tgithub.com/renovate-gomod-issue/bar v1.0.1\n\tgithub.com/renovate-gomod-issue/foo v1.0.0\n)\n\nreplace (\n\tgithub.com/renovate-gomod-issue/bar => ../bar\n\tgithub.com/renovate-gomod-issue/foo => ../foo\n)\n","time":"2021-08-25T14:22:11.337Z","v":0}
{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","cmd":"go","args":"get -d ./...","msg":"go get command included","time":"2021-08-25T14:22:11.342Z","v":0}
{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","command":["go get -d ./..."],"msg":"Executing command","time":"2021-08-25T14:22:11.343Z","v":0}
{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","err":{"killed":false,"code":1,"signal":null,"cmd":"go get -d ./...","stdout":"","stderr":"go: github.com/renovate-gomod-issue/bar@v1.0.1 (replaced by ../bar): reading ../bar/go.mod: open /tmp/renovate/repos/github/renovate-gomod-issue/bar/go.mod: no such file or directory\n","message":"Command failed: go get -d ./...\ngo: github.com/renovate-gomod-issue/bar@v1.0.1 (replaced by ../bar): reading ../bar/go.mod: open /tmp/renovate/repos/github/renovate-gomod-issue/bar/go.mod: no such file or directory\n","stack":"Error: Command failed: go get -d ./...\ngo: github.com/renovate-gomod-issue/bar@v1.0.1 (replaced by ../bar): reading ../bar/go.mod: open /tmp/renovate/repos/github/renovate-gomod-issue/bar/go.mod: no such file or directory\n\n    at ChildProcess.exithandler (child_process.js:319:12)\n    at ChildProcess.emit (events.js:376:20)\n    at ChildProcess.emit (domain.js:470:12)\n    at maybeClose (internal/child_process.js:1055:16)\n    at Socket.<anonymous> (internal/child_process.js:441:11)\n    at Socket.emit (events.js:376:20)\n    at Socket.emit (domain.js:470:12)\n    at Pipe.<anonymous> (net.js:673:12)"},"msg":"rawExec err","time":"2021-08-25T14:22:11.370Z","v":0}
{"name":"renovate","hostname":"localhost.localdomain","pid":479336,"level":20,"logContext":"NcssAJ-sw","repository":"renovate-gomod-issue/fully-replaced","branch":"renovate/github.com-renovate-gomod-issue-bar-1.x","err":{"killed":false,"code":1,"signal":null,"cmd":"go get -d ./...","stdout":"","stderr":"go: github.com/renovate-gomod-issue/bar@v1.0.1 (replaced by ../bar): reading ../bar/go.mod: open /tmp/renovate/repos/github/renovate-gomod-issue/bar/go.mod: no such file or directory\n","message":"Command failed: go get -d ./...\ngo: github.com/renovate-gomod-issue/bar@v1.0.1 (replaced by ../bar): reading ../bar/go.mod: open /tmp/renovate/repos/github/renovate-gomod-issue/bar/go.mod: no such file or directory\n","stack":"Error: Command failed: go get -d ./...\ngo: github.com/renovate-gomod-issue/bar@v1.0.1 (replaced by ../bar): reading ../bar/go.mod: open /tmp/renovate/repos/github/renovate-gomod-issue/bar/go.mod: no such file or directory\n\n    at ChildProcess.exithandler (child_process.js:319:12)\n    at ChildProcess.emit (events.js:376:20)\n    at ChildProcess.emit (domain.js:470:12)\n    at maybeClose (internal/child_process.js:1055:16)\n    at Socket.<anonymous> (internal/child_process.js:441:11)\n    at Socket.emit (events.js:376:20)\n    at Socket.emit (domain.js:470:12)\n    at Pipe.<anonymous> (net.js:673:12)"},"msg":"Failed to update go.sum","time":"2021-08-25T14:22:11.370Z","v":0}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes:

As of now, only inline replace directives are commented out with - "// renovate-comment" before further processing. (lib/manager/gomod/artifacts.ts)

Added the same behavior for the multi-line replace variant.

## Context:

Closes #11428

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/gomod/artifacts.ts [hunk 1]
```diff
@@ -152,10 +152,21 @@ export async function updateArtifacts({
...
+    const inlineReplaceRegEx: RegExp = regEx(
+      /\n(replace\s+[^\s]+\s+=>\s+\.\.\/.*)/g
     );
+
+    const inlineCommentOut = '\n// renovate-replace $1';
+
+    const blockReplaceRegEx: RegExp = regEx(/\nreplace\s*\([^)]+\s*\)/g);
+
+    const blockCommentOut = (match): string =>
+      match.replaceAll('\n', '\n// renovate-replace ');
```
Comment (viceice) [inline lib/manager/gomod/artifacts.ts:164]:
```suggestion
      match.replaceAll('(\r?\n)', '$1// renovate-replace ');
```

Comment (Gabriel-Ladzaretti) [reply lib/manager/gomod/artifacts.ts]: added all the necessary changes as suggested.

does it really matter if we include the \r in the regex, as it is not effected by the regex prior to the suggested changes, as it is untouched if any exists. would love to know as i might have some misconceptions regarding regex :).

**Comment by Gabriel-Ladzaretti** [comment]
im aware that the PR and the commits are made from two different users of mine.
hope it wont make any problems, will avoid this in the future.

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
Can't update branch, did you disabled maintainer access or did you open the PR from an organization?

**Comment by Gabriel-Ladzaretti** [comment]
@viceice "Allow edits by maintainers" was indeed uncheck for some reason.
but also note that the PR was made from a account related to an org, if that what you meant.

**Comment by viceice** [comment]
> @viceice "Allow edits by maintainers" was indeed uncheck for some reason.
> sorry for that.
> please also note that the PR was made from a account related to an org, if that what you meant.
> 

Yes and no. If the fork is in an org, the checkbox isn't available and we maintainer hove no access to the fork branch, so we can't update your branch. As we merge a lot branches a day, your PR would be outdated very often and we need to wait for your update.
😉

**Review by viceice — APPROVED** [review]

**Pull request merged**